bin/src/data/csv.py

```diff
@@ -223,29 +222,48 @@ def get_and_encode(self, dictionary: dict, idx: Any) -> dict:
             # get the data at the given index
             # if the data is not a list, it is converted to a list
             # otherwise it breaks Float().encode_all(data) because it expects a list
-            data = dictionary[key][idx]
+            if idx is None:
```

Comment:
please put data = Something explicitly above the if. I know python can recover lower level indentation variables in higher level indentation even if they were not initialized as such level. But that is just something python should'n't be doing to begin with.

Reply:
why?